Doctrine/Orm/Filter/AbstractFilter.php

```diff
@@ -69,7 +72,111 @@ protected function getClassMetadata(ResourceInterface $resource)
      */
     protected function isPropertyEnabled($property)
     {
-        return null === $this->properties || array_key_exists($property, $this->properties);
+        $enabled = false;
+
+        if (null === $this->properties) {
+            // to ensure sanity, nested properties must still be explicitly enabled
+            $enabled = !$this->isPropertyNested($property);
```

Comment:
I cannot agree that this is cleaner code. The benefit is clear when it's one of the following:

```
    return;
    return false;
    return null;
    return [];
```

but in this case there's actually more work being done.